src/basic/process-util.c

```diff
@@ -1570,13 +1570,17 @@ int setpriority_closest(int priority) {
 }
 
 bool invoked_as(char *argv[], const char *token) {
-        if (!argv || isempty(argv[0]))
+        const char *e = getenv("SYSTEMD_INVOKED_AS");
```

Comment:
getenv_secure()?

also, should this maybe unset the env var, and cache the result? so that this is not propagated further down? maybe via a getenv_secure_steal() helper?

Reply:
> getenv_secure()?

I considered that, but I don't see how it'd be relevant.  This is explicitly for executables, and not library code. Also, similar helpers for other variables don't use "secure_".

Reply:
I dropped the commit to add $SYSTEMD_INVOKED_AS from this PR. I think it's still useful, but it needs more discussion about inheritance, etc, so let's discuss it separately.